src/core/moves/MetropolisHastingsMove.cpp

```diff
@@ -300,11 +381,34 @@ void MetropolisHastingsMove::performMcmcMove( double prHeat, double lHeat, doubl
 
             if ( the_node->isClamped() )
             {
-                ln_likelihood_ratio += the_node->getLnProbabilityRatio();
+                try {
+                    ln_likelihood_ratio += the_node->getLnProbabilityRatio();
+                }
+                catch (RbException e)
+                {
+                    ln_likelihood_ratio = RbConstants::Double::neginf;
+
+                    if ( e.getExceptionType() != RbException::MATH_ERROR )
+                    {
+                        throw RbException(e);
+                    }
+                }
             }
             else
             {
-                ln_prior_ratio += the_node->getLnProbabilityRatio();
+                try
+                {
+                    ln_prior_ratio += the_node->getLnProbabilityRatio();
+                }
+                catch (RbException e)
+                {
+                    ln_prior_ratio = RbConstants::Double::neginf;
+
+                    if ( e.getExceptionType() != RbException::MATH_ERROR )
+                    {
+                        throw RbException(e);
```

Comment:
throw;

Reply:
Does a simple "throw" without giving the exception work here? If I add "throw e;", then I get a warning about throwing a local variable.

Reply:
Changed in previous commit.